quantumdb-core/src/main/java/io/quantumdb/core/versioning/Changelog.java

```diff
@@ -100,9 +100,15 @@ public Changelog addChangeSet(Version appendTo, ChangeSet changeSet, Operation..
 	 */
 	public Changelog addChangeSet(Version appendTo, ChangeSet changeSet, Collection<Operation> operations) {
 		lastAdded = appendTo;
-
+		int size = operations.size();
 		for (Operation operation : operations) {
-			lastAdded = new Version(idGenerator.generateId(), lastAdded, changeSet, operation);
+			size--;
+			if (size == 0) {
+				lastAdded = new Version(changeSet.getId(), lastAdded, changeSet, operation);
```

Comment:
Last version in the changeset gets the id from the changeset. This makes forking much easier as you already know which id it gets.